### PR TITLE
Intermediate.ATX.2017Q1.jeff.slocum.

### PR DESCRIPTION
**Pull request by jeffgojeff**
# Checkpoint Rubric

This is the rubric that your instructor will use to grade your checkpoints. Please do not edit. 

## Checkpoint 1: Towers of Hanoi Terminal App
- [x] All tests passed: 55 points
- [x] Properly commented code: 15 points
- [x] Properly indented code: 15 points
- [x] Demonstrated effective use of JavaScript: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*

## Checkpoint 2: Towers of Hanoi GUI App
- [x] All tests passed: 55 points
- [x] Properly commented code: 15 points
- [x] Properly indented code: 15 points
- [x] Demonstrated effective use of jQuery: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*

## Checkpoint 3: Final Project Gist Blog
- [x] Effective use of AJAX: 20 points
- [ ] Effective use of loops: 20 points
- [x] Effective use of github API: 20 points
- [x] Style and design: 10 points
- [x] Properly commented code: 15 points
- [x] Properly indented code: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*

### CODE REVIEW

File: apps/02RockPaperScissors.js [hunk 1]
```diff
@@ -11,6 +11,37 @@ var rl = readline.createInterface({
 function rockPaperScissors(hand1, hand2) {
 
   // Write code here
+  if (hand1.toLowerCase() === hand2.toLowerCase()){
+    return('It\'s a tie!')
+  }
+  //
```
Comment (cruzmndz) [inline apps/02RockPaperScissors.js:17]: Yep, everything looks copesthetic.

File: apps/03TicTacToe.js [hunk 2]
```diff
@@ -25,24 +33,44 @@ function printBoard() {
...
 
 function diagonalWin() {
   // Your code here
+  if ((board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] === playerTurn) || (board[0][2] === playerTurn && board[1][1] === playerTurn && board[2][0] === playerTurn)){
+
+  return true;
+  }
 }
 
 function checkForWin() {
```
Comment (jw56578) [inline apps/03TicTacToe.js]: This function is missing one piece of required functionality

Comment (jw56578) [inline apps/03TicTacToe.js]: This function is missing one piece of required functionality. Rerun your tests, one is failing.

File: apps/03TicTacToe.js [hunk 3]
```diff
@@ -25,24 +33,45 @@ function printBoard() {
...
+    newGame();
+    return true;
+  }
 }
 
 function ticTacToe(row, column) {
   // Your code here
+  board[row][column] = playerTurn;
+  checkForWin();
+  playerTurn = (playerTurn === 'X') ? 'O' : 'X';
```
Comment (pieroapretto) [inline apps/03TicTacToe.js:70]: Logic checks out!

File: apps/04TowersOfHanoi.js [hunk 4]
```diff
@@ -13,30 +13,108 @@ var stacks = {
...
+  //check for startStack value
+  var cheek = stacks[startStack];
+  var cheek2 = cheek[cheek.length-1];
+  //value check again, commented out when program was finished
+  //console.log('start ', cheek2);
+  
+  //check to see if end spot is bigger that start stop
+  //test for zero, was getting an undefined value here
+  //created zeroCheck to specifically check for zero and not undefined. 
+  if (zeroCheck === 0){
```
Comment (pieroapretto) [inline apps/04TowersOfHanoi.js:59]: You could also pass in `!check2` instead of `zeroCheck` for this if statement on line 59. 

If `check2` is undefined, you know that endStack is empty.

File: apps/04TowersOfHanoi.js [hunk 5]
```diff
@@ -13,30 +13,108 @@ var stacks = {
...
-function checkForWin() {
+function checkForWin(startStack, endStack) {
   // Your code here
 
+  //created chicken game winning print when game is completed.
+  if (stacks.b.length === 4 || stacks.c.length === 4){
+    console.log(" ");
+    console.log("********************************");
+    console.log("*                              *")
+    console.log("*Winner Winner Chicken Dinner!!*");
```
Comment (pieroapretto) [inline apps/04TowersOfHanoi.js:83]: Haha yes!

File: apps/04TowersOfHanoi.js [hunk 6]
```diff
@@ -13,30 +13,108 @@ var stacks = {
...
+
 }
 
 function towersOfHanoi(startStack, endStack) {
   // Your code here
-
+  // check for legality
+  if (isLegal(startStack, endStack)) {
+    movePiece(startStack, endStack);
+    checkForWin(startStack, endStack);
```
Comment (pieroapretto) [inline apps/04TowersOfHanoi.js]: Good syntax and logic flow here.

File: homework/03LessonThree.js [hunk 7]
```diff
@@ -42,7 +42,7 @@ function returnTheFirstArgument(firstArgument) {
 //
 // Your Answer Goes Here:
 //
-//
+//An item's scope refers to where exactly it is accessible from
```
Comment (pieroapretto) [inline homework/03LessonThree.js:45]: Correctness = 100%!

File: homework/03LessonThree.js [hunk 8]
```diff
@@ -80,7 +80,9 @@ var problemSevenAnswer;
 //
 // Your Answer Goes Here:
 //
-//
+// The ternary operator is a simplified version of a quick if else statement.
```
Comment (pieroapretto) [inline homework/03LessonThree.js:83]: Exactly.

File: homework/04LessonFour.js [hunk 9]
```diff
@@ -77,6 +88,7 @@ var nestedObjects = {
   }
 };
 // update the starring property to have the value 'Tim Robbins'
+nestedObjects['users']['bob roberts']['starring'] = 'Tim Robbins';
```
Comment (pieroapretto) [inline homework/04LessonFour.js:91]: nicely done

File: apps/06.1Checkers.js [hunk 10]
```diff
@@ -53,16 +62,139 @@ function Board() {
...
+
+
+    //for loop shit
+    // **IN CLASS**
+    /*for(var i = 0; i < whiteCheckers.length ; i ++){
+      var spot = whiteCheckers[i];
+    }*/
+
+
+    for (var spot of whitePositions){
```
Comment (pieroapretto) [inline apps/06.1Checkers.js:87]: Nice for loop!

File: apps/06.1Checkers.js [hunk 11]
```diff
@@ -53,16 +62,139 @@ function Board() {
...
+     this.board.grid[startRow][startColumn] = null;
+     this.board.grid[endRow][endColumn] = checker;
+     
+     var distance = Math.abs(startRow - endRow);
+     var midRow = (startRow + endRow) / 2;
+     var midCol = (startColumn + endColumn) / 2;
+     var killPosition = [midRow, midCol];
+ 
+     if(distance === 2) {
+       this.board.killChecker(killPosition);
```
Comment (pieroapretto) [inline apps/06.1Checkers.js:193]: All test passed!

wizard

File: apps/06SpaceTravelToMars.js [hunk 12]
```diff
@@ -1,59 +1,103 @@
...
+   commander: 'Main Ship',
+    programmer: 'Any Ship!'
+  };
+  
+ 
+  // Your code here
+ class CrewMember {
+   constructor(name, job, specialSkill) {
+     if(name === '') {
+       return "Please give crew member a name.";
```
Comment (pieroapretto) [inline apps/06SpaceTravelToMars.js:17]: Nice data scrubbing.

File: homework/06LessonSix.js [hunk 13]
```diff
@@ -3,15 +3,19 @@
...
 
 
 // Problem 1:
 // Associative Array Refresh
 // Add properties x: 1, y: 2 to point to make it a point
 // in two dimensional space.
-var point = {
+var point = { 
+  x: 1,
+  y: 2,
```
Comment (pieroapretto) [inline homework/06LessonSix.js:18]: You don't need a comma after your last object attribute (y: 2)

File: homework/06LessonSix.js [hunk 14]
```diff
@@ -36,7 +42,8 @@ var anotherPoint;
 //
 // What is the difference between point (an object literal, or associative array)
 // and anotherPoint (a point object constructed from the Point class)?
-//
+// A point is a class, that other objects can be formed from, with the same 'properties' but with different values.
```
Comment (pieroapretto) [inline homework/06LessonSix.js:45]: Nice answer.

File: homework/06LessonSix.js [hunk 15]
```diff
@@ -96,7 +107,7 @@ var conferenceRoom = new ConferenceRoom();
 // Write your answer in comments
 //
 // What is a method?
-//
+// A method is another word for action that an object can preform.
```
Comment (pieroapretto) [inline homework/06LessonSix.js:110]: Yes. A method is a function assigned to an object as an attribute.

File: apps/07jquery-tic-tac-toe/script.js [hunk 16]
```diff
@@ -2,4 +2,55 @@
...
+    if ($('[data-cell=0]').text() === playerTurn && $('[data-cell=1]').text() === playerTurn && $('[data-cell=2]').text() === playerTurn) {
+      $('#announce-winner').text('Player ' + playerTurn + ' Wins!');
+      return true;  
+    }
+    if ($('[data-cell=3]').text() === playerTurn && $('[data-cell=4]').text() === playerTurn && $('[data-cell=5]').text() === playerTurn) {
+      $('#announce-winner').text('Player ' + playerTurn + ' Wins!');
+      return true;  
+    }
+    if ($('[data-cell=6]').text() === playerTurn && $('[data-cell=7]').text() === playerTurn && $('[data-cell=8]').text() === playerTurn) {
+      $('#announce-winner').text('Player ' + playerTurn + ' Wins!');
```
Comment (pieroapretto) [inline apps/07jquery-tic-tac-toe/script.js:26]: Lots of code duplication handling your winning message. I would recommend calling `$('#announce-winner').text('Player ' + playerTurn + ' Wins!');` once if `checkForWin()` returns true.

File: apps/08jquery-towers-of-hanoi/script.js [hunk 17]
```diff
@@ -1,5 +1,43 @@
...
+    }
+  
+
+  });
+
+  function checkForWin(){
+      //check for win
+      if( $('[data-stack="2"]').children().length === 4 || $('[data-stack="3"]').children().length === 4 ){
+      //check the 2 divs that are correct for winning if they are equal to 4.
+      $('#announce-game-won').text('you won!');
```
Comment (pieroapretto) [inline apps/08jquery-towers-of-hanoi/script.js:38]: all tests passed!

success

File: apps/08jquery-towers-of-hanoi/script.js [hunk 18]
```diff
@@ -1,5 +1,43 @@
...
+      if($blockToBePlaced < $lastBlockValue || !$lastBlockValue) {
+        //comparing the block that was selected, to the last block.
+        
+        $(this).append($block);
+        $block = null;
+        checkForWin();
+      }
+              
+    } else {
+      $block = $(this).children().last().detach();
```
Comment (pieroapretto) [inline apps/08jquery-towers-of-hanoi/script.js:28]: You could also use your predefined $lastBlock variable here if you make it accessible to the entire function scope.

`$block = $lastBlock).detach();`

File: homework/07LessonSeven/script.js [hunk 19]
```diff
@@ -69,20 +69,24 @@ $(document).on('ready', function() {
...
   // and set window.dataIdValue to the value of the data-id attribute of the element
   // you can do this by calling .data() on the selected element
-  window.dataIdValue;
+  window.dataIdValue = $('#data-element').data('id');
+
 
   // Problem 12:
   // select the element with id="data-element"
   // and set its text (.text()) to the value of the data-content attribute
+  var content = $('#data-element').data('content');
```
Comment (pieroapretto) [inline homework/07LessonSeven/script.js:90]: Nicely done.

File: apps/11gist-blog/script.js [hunk 20]
```diff
@@ -1,5 +1,72 @@
...
+          var name = comment.user.login;
+          //assign users comment to a variable.
+          var userComment = comment.body;
+          //makes the users name and there comment a string.
+          var str = (
+            '<li id="user">' + name + '</li>' +
+            '<li id="comment">' + userComment + '</li>'
+          )
+          //adding user's comments to the id #comments.
+          $('#comments').append(str);
```
Comment (jw56578) [inline apps/11gist-blog/script.js:54]: might be some issues here when you click back and forth between posts over and over.
What happens to the comments area
